ecllibrary/std/Date.ecl

```diff
@@ -764,6 +764,34 @@ EXPORT STRING SecondsToString(Seconds_t seconds, VARSTRING format = '%Y-%m-%dT%H
     TimeLib.SecondsToString(seconds, format);
 
 
+/**
+ * Converts a Timestamp_t value into a human-readable string using a format template.
+ *
+ * @param timestamp     The microseconds since epoch.
+ * @param format        The format template to use for the conversion; see
+ *                      strftime() for appropriate format specifiers.  Two
+ *                      additional format specifiers are available to show
+ *                      fractional seconds:
+ *                          %@ - fraction of seconds in microseconds (6 digits)
+ *                          %# - fraction of seconds in milliseconds (3 digits)
+ *                      Millisecond fractions are rounded up from microseconds
+ *                      when necessary.  The maximum length of the resulting
+ *                      string is 255 characters.  This paramter is optional and
+ *                      defaults to '%Y-%m-%dT%H:%M:%S.%@' which is
+ *                      YYYY-MM-DDTHH:MM:SS.ssssss.
+ * @return              The converted timestamp as a string.
+ */
+
+EXPORT STRING TimestampToString(Timestamp_t timestamp, VARSTRING format = '%Y-%m-%dT%H:%M:%S.%@') := FUNCTION
+    f := (UNSIGNED4)(timestamp % 1000000);
+    s1 := TimeLib.SecondsToString(timestamp DIV 1000000, format);
+    s2 := REGEXREPLACE('%@', s1, (STRING)f);
+    s3 := REGEXREPLACE('%#', s2, (STRING)ROUND(f / 1000));
```

Comment:
Is ROUND correct?  999999 rounded would become 1000.  I suspect DIV would be better.